TestRun - Password Criteria Checker/Program.cs

```diff
@@ -0,0 +1,175 @@
+Console.WriteLine("This application will test a given Password against the following criteria \n");
+Console.WriteLine(criteria1);
+Console.WriteLine(criteria2);
+Console.WriteLine(criteria3);   
+Console.WriteLine(criteria4);   
+
+
+// Request user password
+// This space requests a password from the user to be tested against established criteria
+Console.WriteLine("Please submit password for testing against criteria : \n");
+string userPass = Console.ReadLine();
+
+// Criteria Compliance Counter
+// This space initiates variables to count the number of criteria to be adhered to (criteriaCounter), as well as the number of criteria that has been adhered to following each check (complianceCounter).
+int complianceCounter = 0;
+int criteriaCounter = 0;
+
+// Compliance Checker Header
+// Cosmetic output to Console
+Console.WriteLine("");
+Console.WriteLine("-------------------------------------------");
+Console.WriteLine("         PASSWORD CRITERIA CHECKER         ");
+Console.WriteLine("-------------------------------------------");
+Console.WriteLine("");
+
+// criteria1
+// Begins testing of criteria1.
+// Adds 1 to criteriaCounter, indicating that there is now 1 additional rule to comply to.
+criteriaCounter++;
+
+// initiates variables denoting the minimum and maximum number of characters in the given password.
+int passLengthMin = 9;
+int passLengthMax = 25;
+// initiates an integer that measures the length of the given password.
+int passLength = userPass.Length;
+// intiates a boolean that confirms the length of the given password to be greater than or equal to the minimum, and less than or equal to the maximum.
+bool passLengthCheck = (passLength >= passLengthMin && passLength <= passLengthMax);
+// according to resulting true or false regard the last boolean check, outputs to console.
+// if true, adds 1 to complianceCounter to indicate passing score within this criteria
+if (passLengthCheck == true ) {
+    complianceCounter++;
+    Console.WriteLine(criteria1 + " : PASS");
+}
+else 
+{ Console.WriteLine(criteria1 + " : FAIL"); }
+
+// criteria2
+// Begins testing of criteria2.
+// Adds 1 to criteriaCounter, indicating that there is now 1 additional rule to comply to.
+criteriaCounter++;
+// initiates a boolean that confirms if any special characters exist within the given password.
+bool passSpecChar = (
+   userPass.Contains("!") ||
+   userPass.Contains("@") ||
+   userPass.Contains("#") ||
+   userPass.Contains("$") ||
+   userPass.Contains("%") ||
+   userPass.Contains("^") ||
+   userPass.Contains("&") ||
+   userPass.Contains("*") ||
+   userPass.Contains("(") ||
+   userPass.Contains(")") ||
+   userPass.Contains("-") ||
+   userPass.Contains("_") ||
+   userPass.Contains("=") ||
+   userPass.Contains("+") ||
+   userPass.Contains("[") ||
+   userPass.Contains("{") ||
+   userPass.Contains("]") ||
+   userPass.Contains("}") ||
+   userPass.Contains("\\") ||
+   userPass.Contains("|"));

```

Comment:
Look into using regular expressions. They are a powerful tool useful for finding substrings and other string related functions. There are many online previewers available, my favourite is https://regexr.com/

```ruby
/[^\w\d\s]/
```
^ would try to match character that isn't a word (letter), not a digit, or not whitespace. Alternatively, you could specify which characters you want to treat as "special characters"

https://regexr.com/6hv3j